### ISSUE TITLE
Ktor Client Serialization JS: Can't obtain default serializer / NoSuchElementException

### ISSUE THREAD

**Issue by SebastianAigner** [ktorio/ktor#1731]
**Ktor Version and Engine Used (client or server and name)**
```kotlin
kotlinVersion = "1.3.70"
serializationVersion = "0.20.0"
ktorVersion = "1.3.2"
```
For a detailed list of modules, see the minimal example project at https://github.com/SebastianAigner/reproduce-ktor-clients-issue

**Describe the bug**
When initializing a Ktor client on the JS platform [like so](https://github.com/SebastianAigner/reproduce-ktor-clients-issue/blob/master/src/main/kotlin/Main.kt#L15):
```kotlin
val jsonClient = HttpClient {
    install(JsonFeature) {
        serializer = defaultSerializer()
    }
}
```
An exception is thrown:
```
Uncaught NoSuchElementException {message_8yp7un$_0: "List is empty.", cause_th0jdv$_0: null, name: "NoSuchElementException", stack: "NoSuchElementException: List is empty.
at firs…lient-core/1.3.2/ktor-ktor-client-core.js:318:18)"}
```

![image](https://user-images.githubusercontent.com/2178959/76795501-a73b7300-67c9-11ea-9981-c364c4eab185.png)

**To Reproduce**

- Clone https://github.com/SebastianAigner/reproduce-ktor-clients-issue example project
- Run the `browserDevelopmentRun` Gradle task
- Open Chrome inspector. Console will show the error message from above.

Workaround: in `src/main/kotlin/Main.kt`, specify the `serializer` in [line 17](https://github.com/SebastianAigner/reproduce-ktor-clients-issue/blob/master/src/main/kotlin/Main.kt#L17) explictly to be `kotlinx`:
```kotlin
serializer = KotlinxSerializer()
```
After this adjustment, the GET request in the example project works fine.

**Expected behavior**
From the [documentation](https://ktor.io/clients/http-client/features/json-feature.html):
> The JsonFeature has a default serializer(implicitly obtained or by calling defaultSerializer()) [...] based on `kotlinx.serialization` for Native as well as for JavaScript.

I should be able to implictly obtain the `KotlinxSerializer()` and should be able to obtain it via `defaultSerializer()`.

**Comment by e5l** [comment]
It looks like the module was treated as unused and removed by js DCE. @bashor

**Comment by bashor** [comment]
@SebastianAigner have you tried the workaround from [KT-36484](https://youtrack.jetbrains.com/issue/KT-36484)? Does it help?

**Comment by SebastianAigner** [comment]
@bashor yeah, the workaround does help!

**Comment by oleg-larshin** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/KTOR-441">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.

**Comment by e5l** [comment]
Fixed in Kotlin 1.4.0
